tests/scripts/all.sh

```diff
@@ -3448,6 +3448,35 @@ component_test_malloc_0_null () {
     tests/ssl-opt.sh -e 'proxy'
 }
 
+support_test_aesni () {
+    # require an x64_64 target
+    gcc -v 2>&1 | grep Target | grep -q x86_64
+}
+
+component_test_aesni () { # ~ 20s
+    msg "build: default config with MBEDTLS_HAVE_ASM and MBEDTLS_AESNI_C enabled"
+    scripts/config.py set MBEDTLS_AESNI_C
+    scripts/config.py set MBEDTLS_HAVE_ASM
+
+    msg "AES tests, MBEDTLS_AESNI_HAVE_CODE=1 (asm)"
+    make lib tests CC=gcc CFLAGS='-O2 -Werror -Wall -Wextra -DMBEDTLS_AESNI_HAVE_CODE=1'
+    cd tests
+    for t in `find . -type f -executable -name '*aes*'`; do
+        # Run all the suites with aes in their name
+        ./$t
+    done
+    cd ..
+
+    msg "AES tests, MBEDTLS_AESNI_HAVE_CODE=2 (intrinsics)"
```

Comment:
The intention here is to ensure that each AESNI code path (intrinsics, asm) gets tested, not the implementation-selection logic. I've added a plain-C variant as well for completeness.

It doesn't aim to test the intrinsics/asm selection logic. I've prototyped a way to do this but it is a bit clunky (add a uniquely named test symbol to each implementation then scan for it from all.sh with objdump). If you like I can add it to this PR.

Reply:
I'm still not sure what the intended logic is here. What about the case where `MBEDTLS_AESNI_HAVE_CODE=2` (forcing AESNI support via intrinsics to be available) but we aren't building specifically for a platform where this will be supported? Do you consider this to be the default case that's tested by tens of other components? Why are intrinsics and assembly treated differently in this respect? It would help to have a comment explaining the logic of what's tested.